### DISCUSSION TITLE
Replacement for the vm-tools and linux

### DISCUSSION THREAD

**Discussion by EugenMayer** [xenserver/xe-guest-utilities#168] [General]
We are not using Xen Hypervisors ourselfs, but our customers do. So please excuse if my knowledge is lacking :)

We do provide a linux- VM for our customers (appliance) and would like to make it easy for them to install the xen vm tools.

There is the official way documented at https://docs.xenserver.com/en-us/xenserver/8/vms/linux#install-xenserver-vm-tools-for-linux for XENserver (similar for Citrix xen), and there is the one for XCP-ng (https://docs.xcp-ng.org/vms/#linux) also linking here

My question is, if we can use xe-guest-utilities to create an easy-to install package for our debian based VMs, so that customers do not need to download the tools and install it manually in the VM (or we would need to pre-distribute/download the tools, if that is even allowed by the license, since customers are partially in offline-environments)

So hands down, can we utilize xe-guest-utilities to create a offline, on demand installation of the xen-hypervisor tools and if so, for which "distributions is thise supposed to work? Xenserver 8 / XPN-ng or even Citrix Xen 8.2 CU1?

According to https://github.com/xenserver/xe-guest-utilities/releases i seea debian package, so this already looks very promising.

Thanks

**Comment by stormi** [comment]
IIRC, Ubuntu (maybe debian too?) already packages these tools.

FYI there's also a rather well advanced project inside the Xen Project itself, for guest tools that are meant to work with Xen, be it vanilla, in XenServer or in XCP-ng: https://gitlab.com/xen-project/xen-guest-agent

**Reply by EugenMayer** [reply]
thank you

**Reply by EugenMayer** [reply]
Using a fresh debian bookworm container

```
root@0babc4d26923:/# apt search xe-guests-utitlities
Sorting... Done
Full Text Search... Done
```

Neither anything can be found on https://packages.debian.org/search?suite=bookworm&section=all&arch=any&searchon=names&keywords=xe+util

I assume this package is not fitting the debian license model, thus might be available for ubuntu but not debian.

Not a huge issue for us, i can use the debian package provided here without any issues. Just thought to add this here
